### ISSUE TITLE
transition.redirect to named route

### ISSUE THREAD

**Issue by rpkilby** [vuejs/vue-router#203]
`transition.redirect()` only accepts paths. It seems like it would make sense for it to share the same arguments as `router.go()`

**Comment by altitudems** [comment]
I just ran into this limitation as well. Perhaps we could also get the transition.from to be the view we are redirecting from as well?

**Comment by bekite** [comment]
Passing params to transition.redirect does not work
```
transition.redirect({
  name: 'category',
  params: {
    categoryId: category._id
  }
})
```
The params object gets overwritten.
https://github.com/vuejs/vue-router/blob/dev/src/transition.js#L74

**Comment by yyx990803** [comment]
@bekite thanks, fixed in 029a0f7
